staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -659,16 +654,22 @@ func validateCustomResourceDefinitionValidation(customResourceValidation *apiext
 			allowDefaults:            opts.allowDefaults,
 			requireValidPropertyType: opts.requireValidPropertyType,
 		}
+
 		allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISchema(schema, fldPath.Child("openAPIV3Schema"), openAPIV3Schema, true)...)
 
-		if opts.requireStructuralSchema {
+		if opts.requireStructuralSchema && len(allErrs) == 0 {
 			if ss, err := structuralschema.NewStructural(schema); err != nil {
 				// if the generic schema validation did its job, we should never get an error here. Hence, we hide it if there are validation errors already.
```

Comment:
this comment doesn't make sense with `len(allErrs) == 0` moved up

Reply:
before we skipped structural errors if there were non-structurals errors already. I moved it up to apply it to the complete if-else chain.

Reply:
updated the comment